aws/table_aws_s3_object.go

```diff
@@ -0,0 +1,668 @@
+				Hydrate:     getS3Object,
+			},
+			{
+				Name:        "object_parts",
+				Description: "A collection of parts associated with a multipart upload.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ObjectParts"),
+				Hydrate:     getS3ObjectAttributes,
+			},
+			{
+				Name:        "owner",
+				Description: "The owner of the object.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the object.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("TagSet"),
+				Hydrate:     getS3ObjectTagging,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("TagSet").Transform(handleS3TagsToTurbotTags),
+				Hydrate:     getS3ObjectTagging,
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Key"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getObjectARN,
+				Transform:   transform.FromValue().Transform(transform.EnsureStringArray),
+			},
+			{
+				Name:        "region",
+				Description: "The AWS Region in which the object is located.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getBucketLocationForObjects,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+func listS3Objects(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Bucket location will be nil if getBucketLocationForObjects returned an error but
+	// was ignored through ignore_error_codes config arg
+	location, err := getBucketLocationForObjects(ctx, d, h)
+	if err != nil {
+		return nil, err
+	} else if location == "" {
+		return nil, nil
+	}
+
+	svc, err := S3Client(ctx, d, fmt.Sprint(location))
+	if err != nil {
+		plugin.Logger(ctx).Error("listS3Objects", "get_client_error", err)
+		return nil, err
+	}
+
+	bucketName := d.EqualsQuals["bucket_name"].GetStringValue()
+
+	// default supported max value is 1000 by ListObjectsV2
+	maxItems := int32(1000)
+
+	// If the requested number of items is less than the paging max limit
+	// set the limit to that instead
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			maxItems = limit
+		}
+	}
+
+	input := &s3.ListObjectsV2Input{
+		Bucket:     aws.String(bucketName),
+		MaxKeys:    maxItems,
+		FetchOwner: true,
+	}
+
+	equalQuals := d.EqualsQuals
+	if equalQuals["prefix"] != nil {
+		if equalQuals["prefix"].GetStringValue() != "" {
+			input.Prefix = aws.String(equalQuals["prefix"].GetStringValue())
+		}
+	}
+
+	// execute list call
+	objects, err := svc.ListObjectsV2(ctx, input)
```

Comment:
Does the API support pagination?

Reply:
From the API response, I see a parameter `ContinuationToken` and `NextContinuationToken` being returned, could we use it for pagination?

Reply:
yes, I have added the pagination